src/processor/map/map_gds.cpp

```diff
@@ -45,8 +44,10 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapGDSCall(LogicalOperator* logica
     auto sharedState =
         std::make_shared<GDSCallSharedState>(table, std::move(graph), std::move(masks));
     auto printInfo = std::make_unique<GDSCallPrintInfo>(call.getInfo().func.name);
+    auto gdsAlgorithm = call.getInfo().getGDS()->copy();
+    gdsAlgorithm->setSharedState(sharedState);
```

Comment:
OK. Check this again. I did this in gds_call:    
```
void initGlobalStateInternal(ExecutionContext* /*context*/) override {
        gds->setSharedState(sharedState);
    }
```
But this looks wrong because the function is not using its arguments. So this can actually be done during query compilation in map. That's why I had put it there.

Reply:
I see your point. Our table function is executed in the following way

```
tableFunction.exec(localState, sharedState)
```

where both states are stored in the operator and pass as an argument of exec function. In this way we keep the function stateless.

For GDS, since we are setting sharedState to gdsFunction. It's indeed a bit wired. Let keep it as it is and see if we should do the tableFunction way later.

We cannot use `initGlobalStateInternal` because every gdsFunction needs to have its own sharedState being set. `initGlobalStateInternal` will NOT be called on each thread.

Reply:
My tests passed when I used initGlobalStateInternal. I think it's because of a few things. First GDSCall runs single threaded, so we actually don't have the problem of GDSCall running in separate threads. Second, as I understand, initGlobalStateInternal runs once before any worker threads starts copying the physical operator (in this case GDSCall), right? If so, that's also enough because as I was doing before, this setting can even be done in the mapper stage. We don't need worker threads to do this setting.